### ISSUE TITLE
OutOfClassPolynomialStream selects more than active_triples_per_context under tied scores (same class as #150)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#164]
## Problem

At current `main` `da8b86cd33a5982215753fb31bd1b3b6132b3c04`, `OutOfClassPolynomialStream.init` (`alberta_framework/streams/out_of_class.py:195`) selects active oracle triples with the same sort-and-threshold comparison #150 documents for `InteractionFeatureDiscoveryStream`:

```python
threshold = jnp.sort(mask_scores, axis=-1)[..., active_count - 1 : active_count]
mask = mask_scores <= threshold
```

When scores tie at the threshold — a real, reachable condition under `jr.uniform`'s finite float32 representation — every tied element passes `<=`, so a context retains more than `active_triples_per_context` triples. The sqrt-of-count normalization keeps the target scale bounded, but the documented sparsity contract is violated and benchmark comparability between runs that hit or avoid a tie is corrupted — the same argument accepted for #150. The other two streams in the module (`FrequencyMismatchStream`, `CompositionalStream`) do not use this pattern; this is the only remaining site.

## Deterministic falsifier

Identical in shape to #150's: with all scores tied (e.g. a constant score vector) and `active_triples_per_context=2`, the mask retains every triple instead of 2. Reproducible directly by monkeypatching `jr.uniform` to a constant, the same technique PR #151 uses for its regression test.

## Relationship to open work (dedup statement)

#150/#151 (Svector-anu) identify and fix this class in `feature_discovery.py` only — PR #151's diff touches no other file. No open or closed issue or PR references `out_of_class.py`. This issue ports that finding to the remaining site, with credit, the way #109 ported #97's merge guard to the micro sibling.

## Proposed bounded fix

Port #151's exact mechanism: replace the threshold comparison with rank selection —

```python
ranks = jnp.argsort(jnp.argsort(mask_scores, axis=-1), axis=-1)
mask = ranks < active_count
```

— deterministic exact count with index-order tie-breaking. Tie-free selections are unchanged (identical mask for distinct scores). No other behavior, schema, or stream changes.

## Tests first

One regression test in the module's test file mirroring PR #151's: monkeypatch `jr.uniform` to a constant so every score ties, init with `active_triples_per_context=2`, assert exactly 2 nonzero triple weights per task/context — red at the stated baseline (all triples retained), green after the fix. A tie-free invariance pin asserts the selected mask is unchanged for a fixed seed.

## Acceptance

- The new regression test fails at the stated baseline and passes after the fix.
- The module's tests pass; repository Ruff and strict mypy pass; `git diff --check` passes.
- No benchmark seed, learner run, `outputs/` artifact, scientific evidence, or promotion claim is created or modified.

Code PR with failing-test-first evidence follows immediately.
